runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -2823,6 +2823,56 @@ void TR::J9S390JNILinkage::checkException(TR::Node * callNode,
    generateS390LabelInstruction(self()->cg(), TR::InstOpCode::LABEL, callNode, exceptionRestartLabel);
    }
 
+void
+TR::J9S390JNILinkage::processJNIReturnValue(TR::Node * callNode,
+                                            TR::CodeGenerator* cg,
+                                            TR::Register* javaReturnRegister)
+   {
+   auto resolvedMethod = callNode->getSymbol()->castToResolvedMethodSymbol()->getResolvedMethod();
+   auto returnType = resolvedMethod->returnType();
+   bool isUnwrapAddressReturnValue = !((TR_J9VMBase *)fe())->jniDoNotWrapObjects(resolvedMethod)
+                                     && (returnType == TR::Address)
+                                     && (javaReturnRegister!= NULL);
```

Comment:
We also don't need a `javaReturnRegister!= NULL` check here because if the type is address then it must have a return value, and hence a return register. Also we can make this variable assignment `const`.